### ISSUE TITLE
Blank page when click detail trace at Lens Zipkin

### ISSUE THREAD

**Issue by thanhdongnguyen** [openzipkin/zipkin#2829]
I use Zipkin Plugin at Kong Gateway to send data to Zipkin server.
But when i click to detail a trace, it is show blank page and i open Inspect -> Console, it have error: `Uncaught (in promise) TypeError: Cannot read property 'length' of undefined`

This is my json detail of a trace:
```json
[{"traceId":"6cf5624e3452d1bc202a57dad35921c0","parentId":"d65495313f944801","id":"d5cb70daead478dc","name":"kong.header_filter","timestamp":1570005203864000},{"traceId":"6cf5624e3452d1bc202a57dad35921c0","parentId":"d65495313f944801","id":"80b8efd52d66af27","name":"kong.body_filter","timestamp":1570005203864000},{"traceId":"6cf5624e3452d1bc202a57dad35921c0","parentId":"ed395cb866d02665","id":"2a49858b1f1d506b","name":"kong.rewrite","timestamp":1570005203829000},{"traceId":"6cf5624e3452d1bc202a57dad35921c0","parentId":"d65495313f944801","id":"3e53779bdde7c111","name":"kong.access","timestamp":1570005203829000,"duration":16000},{"traceId":"6cf5624e3452d1bc202a57dad35921c0","parentId":"ed395cb866d02665","id":"d65495313f944801","kind":"CLIENT","name":"kong.proxy","timestamp":1570005203829000,"duration":35000,"localEndpoint":{"serviceName":"rewards"},"remoteEndpoint":{"ipv4":"","port":80},"tags":{"kong.route":"aaa20d62-813a-428f-aa40-447324cd30af","kong.service":"872eae11-4571-490b-86d2-59834b93f870","peer.hostname":""}},{"traceId":"6cf5624e3452d1bc202a57dad35921c0","id":"ed395cb866d02665","kind":"SERVER","name":"kong.request","timestamp":1570005203829000,"duration":35000,"remoteEndpoint":{"ipv4":"","port":3771},"tags":{"component":"kong","http.method":"GET","http.status_code":"200","http.url":"","kong.node.id":"37714786-d34b-4513-a459-8020a1f917de"}},{"traceId":"6cf5624e3452d1bc202a57dad35921c0","parentId":"d65495313f944801","id":"862d0f978033cc53","name":"kong.balancer","timestamp":1570005203845000,"remoteEndpoint":{"ipv4":"","port":80},"tags":{"kong.balancer.try":"1"}}]
```

**Comment by codefromthecrypt** [comment]
here's the trace as it comes out from the api.. It is oddly missing durations and quite malformed looking in general. I'll try to figure out what's wrong. 

```json
[
  {
    "traceId": "6cf5624e3452d1bc202a57dad35921c0",
    "parentId": "d65495313f944801",
    "id": "d5cb70daead478dc",
    "name": "kong.header_filter",
    "timestamp": 1570005203864000
  },
  {
    "traceId": "6cf5624e3452d1bc202a57dad35921c0",
    "parentId": "d65495313f944801",
    "id": "80b8efd52d66af27",
    "name": "kong.body_filter",
    "timestamp": 1570005203864000
  },
  {
    "traceId": "6cf5624e3452d1bc202a57dad35921c0",
    "parentId": "ed395cb866d02665",
    "id": "2a49858b1f1d506b",
    "name": "kong.rewrite",
    "timestamp": 1570005203829000
  },
  {
    "traceId": "6cf5624e3452d1bc202a57dad35921c0",
    "parentId": "d65495313f944801",
    "id": "3e53779bdde7c111",
    "name": "kong.access",
    "timestamp": 1570005203829000,
    "duration": 16000
  },
  {
    "traceId": "6cf5624e3452d1bc202a57dad35921c0",
    "parentId": "ed395cb866d02665",
    "id": "d65495313f944801",
    "kind": "CLIENT",
    "name": "kong.proxy",
    "timestamp": 1570005203829000,
    "duration": 35000,
    "localEndpoint": {
      "serviceName": "rewards"
    },
    "remoteEndpoint": {
      "port": 80
    },
    "tags": {
      "kong.route": "aaa20d62-813a-428f-aa40-447324cd30af",
      "kong.service": "872eae11-4571-490b-86d2-59834b93f870",
      "peer.hostname": ""
    }
  },
  {
    "traceId": "6cf5624e3452d1bc202a57dad35921c0",
    "id": "ed395cb866d02665",
    "kind": "SERVER",
    "name": "kong.request",
    "timestamp": 1570005203829000,
    "duration": 35000,
    "remoteEndpoint": {
      "port": 3771
    },
    "tags": {
      "component": "kong",
      "http.method": "GET",
      "http.status_code": "200",
      "http.url": "",
      "kong.node.id": "37714786-d34b-4513-a459-8020a1f917de"
    }
  },
  {
    "traceId": "6cf5624e3452d1bc202a57dad35921c0",
    "parentId": "d65495313f944801",
    "id": "862d0f978033cc53",
    "name": "kong.balancer",
    "timestamp": 1570005203845000,
    "remoteEndpoint": {
      "port": 80
    },
    "tags": {
      "kong.balancer.try": "1"
    }
  }
```

**Comment by codefromthecrypt** [comment]
for example it is using spans instead of annotations for things like "kong.body_filter"

**Comment by codefromthecrypt** [comment]
Notably, there are also spans missing `localEndpoint.serviceName`

Maybe this is something recently fixed in v0.1.3 cc @bungle @jeremyjpj0916

**Comment by codefromthecrypt** [comment]
https://github.com/openzipkin/zipkin/pull/2834 to try to work around the bad data cc @basvanbeek

**Comment by basvanbeek** [comment]
This is one that could originate from Envoy and kill Lens.
```
[
  [
    {
      "traceId": "7f347f457acf182f",
      "id": "7f347f457acf182f",
      "kind": "SERVER",
      "name": "localhost:10000",
      "timestamp": 1570523661459807,
      "duration": 16087,
      "localEndpoint": {
        "ipv4": "169.254.65.45"
      },
      "tags": {
        "component": "proxy",
        "downstream_cluster": "-",
        "guid:x-request-id": "a19160d8-ba84-9c03-ab44-e23b37e4b265",
        "http.method": "POST",
        "http.protocol": "HTTP/1.1",
        "http.status_code": "204",
        "http.url": "http://www.google.com/gen_204?atyp=csi&ei=DUqcXcaVBoaUkwWqjbK4Ag&s=webhp&t=all&bl=Qw3X&imn=2&adh=&conn=onchange&ima=1&imad=0&ime=1&imeb=0&imeo=0&wh=1306&scp=0&net=dl.10000,ect.4g,rtt.50&mem=ujhs.10,tjhs.12,jhsl.4345,dm.8&sto=&sys=hc.12&rt=aft.105,iml.134,prt.117,dcl.121,xjsls.",
        "request_size": "0",
        "response_flags": "-",
        "response_size": "0",
        "upstream_cluster": "service_google",
        "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36"
      },
      "shared": true
    }
  ],
  [
    {
      "traceId": "7955dd635e70bc0a",
      "id": "7955dd635e70bc0a",
      "kind": "SERVER",
      "name": "localhost:10000",
      "timestamp": 1570523661452092,
      "duration": 15314,
      "localEndpoint": {
        "ipv4": "169.254.65.45"
      },
      "tags": {
        "component": "proxy",
        "downstream_cluster": "-",
        "guid:x-request-id": "efcef002-9fb8-91fe-9c8e-dc5805d2dae8",
        "http.method": "POST",
        "http.protocol": "HTTP/1.1",
        "http.status_code": "204",
        "http.url": "http://www.google.com/gen_204?atyp=i&ei=DUqcXcaVBoaUkwWqjbK4Ag&vet=10ahUKEwiG0LnHoIzlAhUGyqQKHaqGDCcQsmQIFg..s&zx=1570523661451",
        "request_size": "0",
        "response_flags": "-",
        "response_size": "0",
        "upstream_cluster": "service_google",
        "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36"
      },
      "shared": true
    }
  ],
  [
    {
      "traceId": "5e918206a4d90f6f",
      "id": "5e918206a4d90f6f",
      "kind": "SERVER",
      "name": "localhost:10000",
      "timestamp": 1570523661417615,
      "duration": 10450,
      "localEndpoint": {
        "ipv4": "169.254.65.45"
      },
      "tags": {
        "component": "proxy",
        "downstream_cluster": "-",
        "guid:x-request-id": "cada645e-f7e8-99e8-9d7b-82ca57f48752",
        "http.method": "GET",
        "http.protocol": "HTTP/1.1",
        "http.status_code": "200",
        "http.url": "http://www.google.com/xjs/_/js/k=xjs.s.en_GB.HUm0oSM0HJs.O/ck=xjs.s.79bvTbAwdes.L.W.O/am=BAAAWATs3YEg_18BACDwCQAAAtyCDRZIh4SKYXVABA/d=1/exm=Fkg7bd,HcFEGb,IvlUe,MC8mtf,OF7gzc,RMhBfe,T4BAC,TJw5qb,TbaHGc,Y33vzc,cdos,csi,d,hsm,iDPoPb,jsa,mvYTse,tg8oTe,uz938c,vWNDde,ws9Tlc,yQ43ff/e",
        "request_size": "0",
        "response_flags": "-",
        "response_size": "47269",
        "upstream_cluster": "service_google",
        "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36"
      },
      "shared": true
    }
  ],
  [
    {
      "traceId": "0a2b5840f7c67e35",
      "id": "0a2b5840f7c67e35",
      "kind": "SERVER",
      "name": "localhost:10000",
      "timestamp": 1570523661308133,
      "duration": 48360,
      "localEndpoint": {
        "ipv4": "169.254.65.45"
      },
      "tags": {
        "component": "proxy",
        "downstream_cluster": "-",
        "guid:x-request-id": "5a9a1297-d06a-9854-bf78-faa80b6a7e77",
        "http.method": "GET",
        "http.protocol": "HTTP/1.1",
        "http.status_code": "200",
        "http.url": "http://www.google.com/xjs/_/js/k=xjs.s.en_GB.HUm0oSM0HJs.O/ck=xjs.s.79bvTbAwdes.L.W.O/m=Fkg7bd,HcFEGb,IvlUe,MC8mtf,OF7gzc,RMhBfe,T4BAC,TJw5qb,TbaHGc,Y33vzc,cdos,hsm,iDPoPb,jsa,mvYTse,tg8oTe,uz938c,vWNDde,ws9Tlc,yQ43ff,d,csi/am=BAAAWATs3YEg_18BACDwCQAAAtyCDRZIh4SKYXVABA/d=1/dg=",
        "request_size": "0",
        "response_flags": "-",
        "response_size": "142355",
        "upstream_cluster": "service_google",
        "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36"
      },
      "shared": true
    }
  ],
  [
    {
      "traceId": "f687e86141d5c18f",
      "id": "f687e86141d5c18f",
      "kind": "SERVER",
      "name": "localhost:10000",
      "timestamp": 1570523661298462,
      "duration": 28971,
      "localEndpoint": {
        "ipv4": "169.254.65.45"
      },
      "tags": {
        "component": "proxy",
        "downstream_cluster": "-",
        "guid:x-request-id": "14193ba0-acaa-91cc-9668-974d81ab91b9",
        "http.method": "GET",
        "http.protocol": "HTTP/1.1",
        "http.status_code": "200",
        "http.url": "http://www.google.com/images/nav_logo299.webp",
        "request_size": "0",
        "response_flags": "-",
        "response_size": "4396",
        "upstream_cluster": "service_google",
        "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36"
      },
      "shared": true
    }
  ],
  [
    {
      "traceId": "737c653fcb0cdf90",
      "id": "737c653fcb0cdf90",
      "kind": "SERVER",
      "name": "localhost:10000",
      "timestamp": 1570523661294615,
      "duration": 14368,
      "localEndpoint": {
        "ipv4": "169.254.65.45"
      },
      "tags": {
        "component": "proxy",
        "downstream_cluster": "-",
        "guid:x-request-id": "23f6d787-e4e3-9d0c-ae69-16375d88fe82",
        "http.method": "POST",
        "http.protocol": "HTTP/1.1",
        "http.status_code": "204",
        "http.url": "http://www.google.com/gen_204?s=webhp&t=aft&atyp=csi&ei=DUqcXcaVBoaUkwWqjbK4Ag&rt=wsrt.119,aft.105,prt.117&bl=Qw3X&ima=1&imad=0&imn=2",
        "request_size": "0",
        "response_flags": "-",
        "response_size": "0",
        "upstream_cluster": "service_google",
        "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36"
      },
      "shared": true
    }
  ],
  [
    {
      "traceId": "1f5ce8075789dfa8",
      "id": "1f5ce8075789dfa8",
      "kind": "SERVER",
      "name": "localhost:10000",
      "timestamp": 1570523661224346,
      "duration": 11862,
      "localEndpoint": {
        "ipv4": "169.254.65.45"
      },
      "tags": {
        "component": "proxy",
        "downstream_cluster": "-",
        "guid:x-request-id": "def6dfe2-0b8c-9a49-8047-91f6536450b1",
        "http.method": "GET",
        "http.protocol": "HTTP/1.1",
        "http.status_code": "200",
        "http.url": "http://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png",
        "request_size": "0",
        "response_flags": "-",
        "response_size": "5969",
        "upstream_cluster": "service_google",
        "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36"
      },
      "shared": true
    }
  ],
  [
    {
      "traceId": "978883983d506fa5",
      "id": "978883983d506fa5",
      "kind": "SERVER",
      "name": "localhost:10000",
      "timestamp": 1570523661059232,
      "duration": 127115,
      "localEndpoint": {
        "ipv4": "169.254.65.45"
      },
      "tags": {
        "component": "proxy",
        "downstream_cluster": "-",
        "guid:x-request-id": "d709b78f-583e-988e-b581-25b72c63852d",
        "http.method": "GET",
        "http.protocol": "HTTP/1.1",
        "http.status_code": "200",
        "http.url": "http://www.google.com/",
        "request_size": "0",
        "response_flags": "-",
        "response_size": "67342",
        "upstream_cluster": "service_google",
        "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36"
      },
      "shared": true
    }
  ]
]
```

**Comment by codefromthecrypt** [comment]
Fixed in 2.17.2 thanks for the details!
